File: SOC/src/cc/game/soc/core/CommodityType.java
package cc.game.soc.core;

public enum CommodityType implements ICardType<DevelopmentArea> {
	Paper("Paper", "Can be exchanged for Science improvements"),
	Cloth("Cloth", "Can be exchanged for Trade improvements"),
	Coin("Coin", "Can be exchanged for Politics improvements");

	final String helpTextId;
	final String nameId;
	DevelopmentArea area;

	CommodityType(String nameId, String helpTextId) {
		this.nameId = nameId;
		this.helpTextId = helpTextId;
	}
	
	@Override
	public CardType getCardType() {
		return CardType.Commodity;
	}

	@Override
	public String getHelpText(Rules rules) {
		return helpTextId;
	}

	@Override
	public DevelopmentArea getData() {
		return area;
	}

	@Override
	public CardStatus defaultStatus() {
		return CardStatus.USABLE;
	}

    @Override
    public String getName() {
        return nameId;
    }


}

File: SOC/src/cc/game/soc/core/CommodityType.kt
package cc.game.soc.core

enum class CommodityType(val _nameId: String, val helpTextId: String) : ICardType<DevelopmentArea> {
	Paper("Paper", "Can be exchanged for Science improvements"),
	Cloth("Cloth", "Can be exchanged for Trade improvements"),
	Coin("Coin", "Can be exchanged for Politics improvements");

    val area: DevelopmentArea
        get() = when (this) {
    		Paper -> DevelopmentArea.Science
		    Cloth -> DevelopmentArea.Trade
		    Coin -> DevelopmentArea.Politics
    	}

	override val cardType = CardType.Commodity

	override fun getHelpText(rules: Rules): String {
		return helpTextId
	}

	override fun getData(): DevelopmentArea = area

	override fun defaultStatus(): CardStatus {
		return CardStatus.USABLE
	}

	override fun getNameId(): String = _nameId;
}